### PR TITLE
Feat/Author sidebar

### PR DESCRIPTION
**Pull request by pitb2022**
https://github.com/thematters/matters-web/issues/4001
https://github.com/thematters/developer-resource/issues/334

### CODE REVIEW

File: src/common/utils/route.ts [hunk 1]
```diff
@@ -115,6 +116,9 @@ export const toPath = (
             mediaHash ? '-' + mediaHash : ''
           }`
         }
+        if (!!args.collectionId) {
+          href = `${href}?cid=${args.collectionId}`
```
Comment (robertu7) [inline src/common/utils/route.ts:120]: "cid" is also referred to https://docs.ipfs.tech/concepts/content-addressing/ under the context of Matters, can use another name (e.g. "collection")

File: src/common/utils/analytics.ts [hunk 2]
```diff
@@ -249,6 +249,7 @@ type ArticleFeedType =
   | 'wallet'
   | 'related_donations'
   | 'circle_detail'
+  | 'article_detail_author-sidebar-collection'
```
Comment (robertu7) [inline src/common/utils/analytics.ts:252]: `article_detail_collection` is more concise?

File: src/components/ArticleDigest/AuthorSidebar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import { ArticleDigestTitle, ArticleDigestTitleTextSize } from '../Title'
+import Placeholder from './Placeholder'
+import styles from './styles.module.css'
+
+export type ArticleDigestAuthorSidebarProps = {
+  article: ArticleDigestAuthorSidebarArticleFragment
+  collectionId?: string
+  titleTextSize?: ArticleDigestTitleTextSize
+  titleColor?: 'greyDarker' | 'black'
+  imageSize?: 44 | 64
```
Comment (robertu7) [inline src/components/ArticleDigest/AuthorSidebar/index.tsx:18]: try not to expose the exact size: `coverSize?: 'sm' | 'md'`?

File: src/components/CollectionDigest/AuthorSidebar/index.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  ResponsiveImage,
+  TextIcon,
+} from '~/components'
+import { CollectionDigestAuthorSidebarCollectionFragment } from '~/gql/graphql'
+
+import styles from './styles.module.css'
+
+export type CollectionDigestAuthorSidebarProps = {
+  collection: CollectionDigestAuthorSidebarCollectionFragment
+  imageSize?: 64
```
Comment (robertu7) [inline src/components/CollectionDigest/AuthorSidebar/index.tsx:17]: same as above, better not to expose the exact size

File: src/views/ArticleDetail/AuthorSidebar/Collection/index.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+      </LinkWrapper>
+      {collection.description && (
+        <LinkWrapper {...collectionDetailPath}>
+          <section className={styles.description}>
+            {collection.description}
+          </section>
+        </LinkWrapper>
+      )}
+      <section className={styles.feed}>
+        <InfiniteScroll
```
Comment (robertu7) [inline src/views/ArticleDetail/AuthorSidebar/Collection/index.tsx:96]: current implementation won't scroll to the position of current article, please confirm with product team for workaround.

File: src/views/ArticleDetail/AuthorSidebar/Tabs/styles.module.css [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      'clig' off,
+      'liga' off;
+    line-height: 1.375rem;
+    color: var(--color-grey-darker);
+    cursor: pointer;
+    background: var(--color-grey-lighter);
+    border-radius: 0.5rem;
+    transition-property: background-color, color;
+
+    /* FIX: change the font-weight of a font, the text will typically cause a bit of a layout shift
```
Comment (robertu7) [inline src/views/ArticleDetail/AuthorSidebar/Tabs/styles.module.css:30]: and the animation when switching between tabs is laggy

**Review by robertu7 — APPROVED** [review]

File: src/common/utils/analytics.ts (hunk 2 above)
Comment (pitb2022) [reply src/common/utils/analytics.ts:252]: Related articles also use the word Collection, which can lead to misunderstandings

**Pull request merged**